Extension/src/LanguageServer/client.ts

```diff
@@ -4114,4 +4145,5 @@ class NullClient implements Client {
     setShowConfigureIntelliSenseButton(show: boolean): void { }
     addTrustedCompiler(path: string): Promise<void> { return Promise.resolve(); }
     getIncludes(): Promise<GetIncludesResult> { return Promise.resolve({} as GetIncludesResult); }
+    getChatContext(token: vscode.CancellationToken): Promise<ChatContextResult> { return Promise.resolve({} as ChatContextResult); }
```

Comment:
(Moved this over from the other PR)

>> Should this return Promise<ChatContextResult | undefined> to match the other definition?

> Could you advice what changes are needed?

The line current is:

```
getChatContext(token: vscode.CancellationToken): Promise<ChatContextResult> { return Promise.resolve({} as ChatContextResult); }
```

But the interface declares it as:

```
getChatContext(token: vscode.CancellationToken): Promise<ChatContextResult | undefined>;
```

I'm just suggesting updating the derived class member function declaration be consistent with the base class, like other functions here.  So:

```
getChatContext(token: vscode.CancellationToken): Promise<ChatContextResult | undefined> { return Promise.resolve({} as ChatContextResult); }
```

Reply:
@Colengms what about dropping undefined everywhere? That is what I've actually done.

Reply:
I think this is a leftover from when the native side was still returning optional values. Removing `undefined` everywhere should be the correct change.

Reply:
done